dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2625,190 @@ void FileSprayer::pushParts()
+    buildAzureStorageURIBase(sourcePlane, sourceBaseURI);
+    sourcePlane->querySASToken(sourceSAStoken);
+    const char * sourcePrefix = sourcePlane->queryPrefix();
+    unsigned sourcePrefixLength = isEmptyString(sourcePrefix)?0:strlen(sourcePrefix);
+
+    // Target
+    StringBuffer targetClusterName, targetBaseURI, targetSAStoken;
+    distributedTarget->getClusterName(0, targetClusterName);
+    Owned<IStoragePlane> targetPlane = getDataStoragePlane(targetClusterName.str(), false);
+    buildAzureStorageURIBase(targetPlane, targetBaseURI);
+    targetPlane->querySASToken(targetSAStoken);
+    const char * targetPrefix = targetPlane->queryPrefix();
+    unsigned targetPrefixLength = isEmptyString(targetPrefix)?0:strlen(targetPrefix);
+    auto targetType = targetPlane->getStorageType();
+
+    std::vector<Shares::StartFileCopyOperation> fileCopyOp;
+    std::vector<StartBlobCopyOperation> blobCopyOp;
+    std::vector<unsigned> copyingInProgress;
+    if (targetType==StorageType::StorageTypeAzureFile)
+        fileCopyOp.reserve(partition.length());
+    else
+        blobCopyOp.reserve(partition.length());
+    // Submit 'copy' operations
+    try
+    {
+         ForEachItemIn(idx, partition)
+         {
+            PartitionPoint & cur = partition.item(idx);
+            OutputProgress & curprogress= progress.item(idx);
+
+            if (curprogress.status==OutputProgress::StatusCopied)
+                continue;
+
+            StringBuffer sourceURI(sourceBaseURI);
+            StringBuffer sourcePath;
+            cur.inputName.getPathInStoragePlane(sourcePath);
+            StringBuffer tmp;
+            sourceURI.appendf("/%s/%s",  encode_url(tmp, sourcePath.str()).str(), sourceSAStoken.str());
+
+            StringBuffer targetURI(targetBaseURI);
+            StringBuffer targetPath;
+            cur.outputName.getPathInStoragePlane(targetPath);
+            targetURI.appendf("/%s/%s", encode_url(tmp.clear(), targetPath.str()).str(), targetSAStoken.str());
+            if (targetType==StorageType::StorageTypeAzureFile)
+            {
+                auto fileClient = Shares::ShareFileClient(targetURI.str());
+                fileCopyOp.emplace_back(std::move(fileClient.StartCopy(sourceURI.str())));
+            }
+            else
+            {
+                auto blobClient = BlobClient(targetURI.str());
+                blobCopyOp.emplace_back(std::move(blobClient.StartCopyFromUri(sourceURI.str())));
+            }
+            copyingInProgress.emplace_back(idx);
+            curprogress.status =  OutputProgress::StatusBegin;
+            curprogress.numReads++;
+            curprogress.numWrites++;
+            if (isAborting())
+                break;
+         }
+    }
+    catch (const Azure::Core::RequestFailedException& e)
+    {
+        IERRLOG("transferUsingAPI failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
```

Comment:
What about copies that have been started?  Should it wait for completion and mark them as copied.  If not will it cause problems trying to overwrite files that already exist.